week2/w2_Node.h

```diff
@@ -0,0 +1,17 @@
+//
+// Created by julia on 5/24/17.
+//
+
+#ifndef WEEK_2_W2_NODE_H
+#define WEEK_2_W2_NODE_H
+
+
+struct Node {
```

Comment:
This is risky. Never trust the users of your code :) 
Ideally, you either make Node and implementation detail of BinaryTree, and don't expose it on it's interface (so all calls take/return the int keys) OR have it's members, constructors and modifiers be private and add the BinaryTree method that needs changing them as friend.